lib/alchemy/resource.rb

```diff
@@ -159,19 +170,24 @@ def editable_attributes
       attributes.reject { |h| restricted_attributes.map(&:to_s).include?(h[:name].to_s) }
     end
 
-    # Returns all columns that are searchable
+    # Returns all attribute names that are searchable in the admin interface
     #
-    def searchable_attributes
-      attributes.select { |a| searchable_attribute?(a) }
-        .concat(searchable_relation_attributes?(attributes))
+    def searchable_attribute_names
+      if model.respond_to?(:searchable_alchemy_attributes)
+        model.searchable_alchemy_attributes
```

Comment:
To be consistent to all other methods of this like, we should rename this to `searchable_alchemy_resource_attributes`. This is longer, but reflects more what it's for: Alchemy resources and it follows the naming of the other methods.